win/WIN_EditTextbox.h

```diff
@@ -27,12 +27,12 @@ namespace win
 		void click();
 
 		void editText(const char* newText);
-		virtual void updateAndRerender(std::string newString);
+		virtual void editTextAndRerender(std::string newString);
```

Comment:
So this is probably best acheived by using a dirty flag: when the text changes, the dirty flag is set, so when you come to render it, the relevant changes can be made (usually in the draw() method). This will probably simplify the public API, which is good.
**SHOULD**

Reply:
removed the 'draw(), SDL present' entirely - realised that this will already be rerendering thanks to changes to events.
This is because mouseMove, mouseButtonUp, ect, now return true/false, and if true then is set to rerender. Should only return true if results in a visual change.
There should be no need for a dirty flag flag if the 'rerender check' under PAINT_program remains the same.